lib/sycamore/sycamore/transforms/sort.py

```diff
@@ -0,0 +1,57 @@
+from typing import Any, Optional
+
+from sycamore.plan_nodes import Node, Transform
+from sycamore.data import Document
+
+from ray.data import Dataset
+
+
+class Sort(Transform):
+    """
+    Sort by field in Document
+    """
+
+    def __init__(self, child: Node, descending: bool, field: str, default_val: Optional[Any] = None):
+        super().__init__(child)
+        self._descending = descending
+        self._field = field
+        self._default_val = default_val
+
+    def execute(self, **kwargs) -> "Dataset":
+        # creates dataset
+        ds = self.child().execute(**kwargs)
+
+        # adds a "key" column containing desired field
+        map_fn = self.make_map_fn_sort()
+        ds = ds.map(map_fn)
+
+        # sorts the dataset
+        ds = ds.sort("key", descending=self._descending)
```

Comment:
I think your implementation is fine, but I do have questions about what this guarantees downstream. Are all of our other transformations order preserving (I think probably)? From the docs, it looks like sort will materialize the dataset in Ray's objet store. Not sure what implications that has for subsequent operations and when it gets freed.

Reply:
Hmm that makes sense to consider, we can talk about it!